### DISCUSSION TITLE
[Solved] How to use groupby, alias and aggregate in REST request?

### DISCUSSION THREAD

**Discussion by vasani-arpit** [directus/directus#8255] [Archive]
I am on directus v9.0.0-rc.93. I want to use aggregation, grouping and alias in API request.

The thing is I am doing as per the doc and none of them are working individually or all together.

Here is my postman request.

![postman request](https://i.stack.imgur.com/IqnGX.png)

In the response I am getting array of the object containing userid, earnedpoints cardcolor only nothing else. 🤷‍♂️

**response**
```json
{
    "meta": {
        "total_count": 9,
        "filter_count": 8
    },
    "data": [
        {
            "userid": 1,
            "earnpoints": 1000,
            "cardcolor": "#E35169"
        },
        {
            "userid": 1,
            "earnpoints": 1000,
            "cardcolor": "#F7971C"
        },
        {
            "userid": 1,
            "earnpoints": 1000,
            "cardcolor": "#9E8DE4"
        },
        {
            "userid": 1,
            "earnpoints": 10,
            "cardcolor": null
        },
        {
            "userid": 1,
            "earnpoints": 5,
            "cardcolor": null
        },
        {
            "userid": 1,
            "earnpoints": 10,
            "cardcolor": null
        },
        {
            "userid": 1,
            "earnpoints": 5,
            "cardcolor": null
        },
        {
            "userid": 1,
            "earnpoints": 3505,
            "cardcolor": null
        }
    ]
}
```

I am not getting any other columns like alias which I set nor I am getting the result of aggregation function.

🙏 Thanks in Advance  !!

**Comment by rijkvanzanten** [comment]
> I am on directus v9.0.0-rc.93. I want to use aggregation, grouping and alias in API request.

Aggregation, grouping, and aliasing was added in RC.94 🙂

**Reply by vasani-arpit** [reply]
Thanks !! I will update and check it out.

**Reply by joselcvarela** [reply]
You must be a wizard 🧙‍♂️
RC94 was just released yesterday with that feature 😁

**Reply by vasani-arpit** [reply]
I checked the document it was updated, so I took the latest at that time which was 93. I suggest we should add info bubble or add a global dropdown for version in top bar.

**Reply by rijkvanzanten** [reply]
Whatever docs to go `main` will update the website. There's a small chance the docs are updated right before the release is made.

The docs in your app are always locked to the version you're on, so those are versioned by default 👍🏻

**Reply by shashwattpandeyy** [reply]
How to get the count in a custom endpoint like meta is not working when I used and I didn't find how to use aggregation in custom endpoint?

**Reply by rijkvanzanten** [reply]
Heya! You can use aggregation in the same way you would in the query parameters:

```js
const { ItemsService } = services;

const myService = new ItemsService('my-collection', { schema });

const aggregate = await myService.readByQuery({
  aggregate: {
    avg: ['value']
  }
});
```

**Reply by rashidpathiyil** [reply]
> > I am on directus v9.0.0-rc.93. I want to use aggregation, grouping and alias in API request.
> 
> Aggregation, grouping, and aliasing was added in RC.94 🙂

Is it possible to get user object ```first_name and last_name``` instead just ```user_id``` when using ```groupBy: "user_created"```

**Reply by benhaynes** [reply]
Yes, just use the `fields` parameter: https://docs.directus.io/reference/query/#fields

`user_created.*` (get all user fields) or `user_created.first_name,user_created.last_name`

So:

```
?fields=*,user_created.first_name,user_created.last_name
```

Will get all parent fields, and the two user_created ones you mentioned.

**Reply by rijkvanzanten** [reply]
No. When you group on a value, you can't relationally nest things under that value. The fields parameter doesn't apply when you aggregate values together 👍🏻

**Comment by Naman-2001** [comment]
I want to get the id of the records with minimum price . But I am only able to retrieve price and product_id fields.
`?aggregate[min]=price&groupBy[]=product_id&fields=product_id,revenue,id`

Response - 
`{
    "data": [
        {
            "product_id": 676,
            "min": {
                "price": 400
            }
        }
    ]
}`

**Reply by rijkvanzanten** [reply]
With the aggregate you're asking "What is the minimum price in this dataset". With the grouping you're saying "What's the minimum price per product_id". What you're looking for is most likely just a regular request sorted by price, something like `/items/products?limit=5&sort=price`
